crates/vault-registry/src/types.rs

```diff
@@ -102,19 +104,242 @@ pub type DefaultVaultId<T> = VaultId<<T as frame_system::Config>::AccountId, Cur
 
 pub type DefaultVaultCurrencyPair<T> = VaultCurrencyPair<CurrencyId<T>>;
 
+pub mod v2 {
+    use super::*;
+
+    #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default, TypeInfo)]
+    struct WalletV2 {
+        // store all addresses for `report_vault_theft` checks
+        pub addresses: BTreeSet<BtcAddress>,
+        // we use this public key to generate new addresses
+        pub public_key: BtcPublicKey,
```

Comment:
Actually I think we should remove this and only use the main key to generate new addresses.